src/clang-cpp-frontend/clang_cpp_convert.cpp

```diff
@@ -1093,35 +1107,9 @@ bool clang_cpp_convertert::get_decl_ref(
     // first follow the base conversion flow to fill new_expr
     if(clang_c_convertert::get_decl_ref(decl, new_expr))
       return true;
-
-    /*
-     * Name and id might be empty.
-     * We need to do some additional checks and conversions on function
-     * parameters for C++, e.g. unnamed const ref in a defaulted cpyctor
-     */
-    const clang::ParmVarDecl &param =
-      static_cast<const clang::ParmVarDecl &>(decl);
-    const clang::DeclContext *dcxt = param.getParentFunctionOrMethod();
-
-    /*
-     * try to get its name and id.
-     * If empty, we need to manually assign a name and an id.
-     */
-    get_decl_name(param, name, id);
-    if(
-      (id.empty() || name.empty()) && is_cpyctor(dcxt) &&
-      is_defaulted_ctor(dcxt))
-    {
-      get_cpyctor_name(
-        static_cast<const clang::CXXConstructorDecl *>(dcxt),
-        id,
-        name,
-        new_expr);
-
-      // assign a name and an id
-      new_expr.name(name);
-      new_expr.identifier(id);
-    }
+    const auto *param = llvm::dyn_cast<const clang::ParmVarDecl>(&decl);
```

Comment:
assert(param); since we kinda assumed this dyn_cast would work and just used it in the code below.